cdn_lambda/functions/map_to_s3.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+
+from chexus import Client, TableItem
+
+TABLE_NAME = "project-exodus-prototyping"
+TABLE_REGION = "us-east-2"
+
+
+def lambda_handler(event, context):
+    _ = context
+
+    # Get uri from origin request event
+    request = event["Records"][0]["cf"]["request"]
+
+    client = Client()
+
+    # Get DB table items with matching web_uri key
+    search_results = client.search(
```

Comment:
This doesn't quite seem right because we're searching for *all* matches on the URI, and then filtering client-side for the right one to use.  Can't the DB do such filtering?  Over time, there will be hundreds to thousands of items for the same URI (thinking of files updated with high frequency like repomd.xml).

As a side note, the DB limits for fetching are not very high (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-api). I'm not sure what operation we are doing here but it would either be BatchGetItem (100 items) or Query (1MB) I guess, both of which it seems we could hit if doing an insufficiently filtered query.  I haven't looked into whether you already added something to chexus for pagination.

Reply:
Yes, I agree. This is just the way it would have to be implemented if using chexus in it's current state, as it only queries exact matches.

I'll build a boto3 query here to retrieve only the latest item.